typeql/language/rule-validation.feature

```diff
@@ -268,6 +269,37 @@ Feature: TypeQL Rule Validation
       """
 
 
+  Scenario: the variables in the conclusion of a rule must be bound in all branches in its 'when' clause
+    Then typeql define; throws exception
+      """
+      define
+      person plays both-named-robert:named-robert;
+      both-named-robert sub relation, relates named-robert;
+      rule typo-in-two-roberts-are-both-named-robert: when {
+        $p isa person, has name "Robert";
+        {$p1 isa person, has name "Robert";} or {$p2 isa person, has name "Bob";};
+      } then {
+        (named-robert: $p, named-robert: $p2) isa both-named-robert;
+      };
+      """
+
+
+  Scenario: variables in a rule's 'then' need not be in the trunk of the 'when' as long as they appear in all branches
+    Then typeql define
+      """
+      define
+      person plays both-named-robert:named-robert;
+      both-named-robert sub relation, relates named-robert;
+      rule two-roberts-are-both-named-robert: when {
+        {$p1 isa person, has name "Robert";} or {$p1 isa person, has name "Bob";};
+        {$p2 isa person, has name "Robert";} or {$p2 isa person, has name "Bob";};
+      } then {
+        (named-robert: $p1, named-robert: $p2) isa both-named-robert;
+      };
+      """
+    Then transaction commits
```

Comment:
you're missing one case where you need to ban a negation containing disjunctions in rules

Reply:
Added above, but supporting instead of banning.